src/resdk/tables/qc.py

```diff
@@ -0,0 +1,959 @@
+        "type": "Int64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "READ_PAIR_OPTICAL_DUPLICATES",
+        "slug": "picard_read_pair_optical_duplicates",
+        "type": "Int64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "PERCENT_DUPLICATION",
+        "slug": "picard_percent_duplication",
+        "type": "Float64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "ESTIMATED_LIBRARY_SIZE",
+        "slug": "picard_estimated_library_size",
+        "type": "Int64",
+        "agg_func": "mean",
+    },
+]
+
+PICARD_INSERT_SIZE_MAP = [
+    {
+        "name": "MEDIAN_INSERT_SIZE",
+        "slug": "picard_insert_median_size",
+        "type": "Float64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "MODE_INSERT_SIZE",
+        "slug": "picard_insert_mode_size",
+        "type": "Float64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "MEDIAN_ABSOLUTE_DEVIATION",
+        "slug": "picard_insert_mad",
+        "type": "Float64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "MIN_INSERT_SIZE",
+        "slug": "picard_insert_min_size",
+        "type": "Float64",
+        "agg_func": "min",
+    },
+    {
+        "name": "MAX_INSERT_SIZE",
+        "slug": "picard_insert_max_size",
+        "type": "Float64",
+        "agg_func": "max",
+    },
+    {
+        "name": "MEAN_INSERT_SIZE",
+        "slug": "picard_insert_mean_size",
+        "type": "Float64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "STANDARD_DEVIATION",
+        "slug": "picard_insert_std_dev",
+        "type": "Float64",
+        "agg_func": "mean",
+    },
+    {
+        "name": "READ_PAIRS",
+        "slug": "picard_insert_read_pairs",
+        "type": "Int64",
+        "agg_func": "sum",
+    },
+    {
+        "name": "PAIR_ORIENTATION",
+        "slug": "picard_insert_pair_orientation",
+        "type": "string",
+        "agg_func": "unique",
+    },
+]
+
+
+def general_multiqc_parser(file_object, name, column_names):
+    """General parser for MultiQC files."""
+    df = pd.read_csv(file_object, sep="\t", index_col=0)
+
+    # Keep only specified columns:
+    df = df[
+        [
+            column.get("name", "")
+            for column in column_names
+            if column.get("name", "") in df.columns
+        ]
+    ]
+    # Rename
+    df = df.rename(
+        columns={
+            column.get("name", ""): column["slug"]
+            for column in column_names
+            if column.get("name", "") in df.columns
+        }
+    )
+
+    if df.empty:
+        return pd.Series(name=name)
+
+    # Perform aggregation
+    series = df.agg(
+        {
+            column["slug"]: column["agg_func"]
+            for column in column_names
+            if column["slug"] in df.columns
+        }
+    )
+    series.name = name
+    return series
+
+
+def macs_prepeak_parser(file_object, name, column_names, sample_name):
```

Comment:
A lot of the code repeats in all parser functions. For the sake of simplicity and speed of merging this PR we can leave it as is, but if/when the number of files we wish to parse expands even more we will need to consider how to abstract/ generalize this.